test/libsolidity/smtCheckerTests/abi/abi_decode_simple.sol

```diff
@@ -21,14 +21,23 @@ contract C {
 // SMTEngine: all
 // SMTIgnoreCex: yes
 // ----
-// Warning 1218: (505-519): CHC: Error trying to invoke SMT solver.
-// Warning 1218: (538-552): CHC: Error trying to invoke SMT solver.
-// Warning 6328: (209-223): CHC: Assertion violation happens here.
-// Warning 6328: (260-274): CHC: Assertion violation happens here.
-// Warning 6328: (359-373): CHC: Assertion violation happens here.
-// Warning 6328: (392-406): CHC: Assertion violation happens here.
-// Warning 6328: (425-434): CHC: Assertion violation happens here.
+// Warning 6328: (191-205): CHC: Assertion violation might happen here.
+// Warning 6328: (209-223): CHC: Assertion violation might happen here.
+// Warning 6328: (242-256): CHC: Assertion violation might happen here.
+// Warning 6328: (260-274): CHC: Assertion violation might happen here.
+// Warning 6328: (359-373): CHC: Assertion violation might happen here.
+// Warning 6328: (392-406): CHC: Assertion violation might happen here.
+// Warning 6328: (425-434): CHC: Assertion violation might happen here.
 // Warning 6328: (505-519): CHC: Assertion violation might happen here.
 // Warning 6328: (538-552): CHC: Assertion violation might happen here.
-// Warning 4661: (505-519): BMC: Assertion violation happens here.
-// Warning 4661: (538-552): BMC: Assertion violation happens here.
+// Warning 3996: CHC analysis was not possible. No Horn solver was available. None of the installed solvers was enabled.
+// Warning 7812: (191-205): BMC: Assertion violation might happen here.
+// Warning 7812: (209-223): BMC: Assertion violation might happen here.
+// Warning 7812: (242-256): BMC: Assertion violation might happen here.
+// Warning 7812: (260-274): BMC: Assertion violation might happen here.
+// Warning 7812: (359-373): BMC: Assertion violation might happen here.
+// Warning 7812: (392-406): BMC: Assertion violation might happen here.
+// Warning 7812: (425-434): BMC: Assertion violation might happen here.
+// Warning 7812: (505-519): BMC: Assertion violation might happen here.
+// Warning 7812: (538-552): BMC: Assertion violation might happen here.
+// Warning 8084: BMC analysis was not possible. No SMT solver (Z3 or CVC4) was available. None of the installed solvers was enabled.
```

Comment:
Where did these changes some from? These look like they're related to SMTChecker but this PR does not touch it at all.

@leonardoalt These aren't legit expectation updates, are they?

Reply:
```
 // Warning 3996: CHC analysis was not possible. No Horn solver was available. None of the installed solvers was enabled. 
  // Warning 8084: BMC analysis was not possible. No SMT solver (Z3 or CVC4) was available. None of the installed solvers was enabled.
```

these are there so looks like @matheusaaguiar just forgot to use `--no-smt` when running the tests

Reply:
Should I run the tests with --no-smt?

Reply:
Ah, right.

@matheusaaguiar Only if you don't have Z3 and CVC4 installed. You need them to recreate the same results in tests.

Reply:
If you don't have latest z3 installed yes

Reply:
But also if your change doesn't affect SMT tests it also makes your tests much faster

Reply:
Ok, running with --no-smt now.